### PR TITLE
99 add flight accordion

### LINKED ISSUE #99: Add Flight info dropdown/accordion
The Upcoming booking page is ready and already has an (incomplete) Accordion component. So we can add the flight info once the design is added to Zeplin. @MRTNKCR @mjancik

Marvel link: https://marvelapp.com/7iihb90/screen/39641486

**Update:** zeplin link: https://zpl.io/adMXoOp

### PR DESCRIPTION
**Pull request by git-toni**
Closes #99 
<br/><br/><br/><url>LiveURL: https://kiwicomsmart-faq-kfogismpnp.now.sh
</url>
</url>
</url>

### CODE REVIEW

File: src/SingleBookingPage/bookingItem/AccordionHeader.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    justify-content: center;
+    cursor: pointer;
+  }
+  .itinerary .location {
+    margin-right: 4px;
+  }
+`;
+type Props = {|
+  trip: AccordionTripSummary_trip,
+  toggle: () => void,
```
Comment (jaroslav-kubicek) [inline src/SingleBookingPage/bookingItem/AccordionHeader.js]: IMHO the whole AccordionHeader should be clickable and this component should accept expanded true/false for chevron icon

Comment (git-toni) [reply src/SingleBookingPage/bookingItem/AccordionHeader.js]: I agree, whole AccordionHeader would be more comfortable. However let's make sure with @MRTNKCR @mjancik's opinoin :slightly_smiling_face: 

> this component should accept expanded true/false for chevron icon

Do you mean the parent `Accordion` component should hold the `toggled/untoggled` state and pass it as props to `AccordionHeader`? If so, I agree, it makes sense conceptually for `Acoordion` to hold the state.

Comment (jaroslav-kubicek) [reply src/SingleBookingPage/bookingItem/AccordionHeader.js]: > I agree, whole AccordionHeader would be more comfortable. However let's make sure with @MRTNKCR @mjancik's opinoin

yes, make the whole header clickable

then you don't need to pass down `toggle` callback and `expanded=true/false` should be enough

File: src/SingleBookingPage/bookingItem/AccordionBody.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+type Props = {|
+  legs: AccordionBody_legs,
+|};
+
+class AccordionBody extends React.Component<Props> {
+  renderLegs = () => {
+    const { legs } = this.props;
+    const lastLeg = legs.slice(-1)[0];
+    if (legs.length > 1) {
```
Comment (jaroslav-kubicek) [inline src/SingleBookingPage/bookingItem/AccordionBody.js]: what about just using map function `.map(leg, index, legs) => { 
  if (index === legs.length - 1) {
    return <AccordionBodyLastLeg />; 
  }

  return <AccordionBodyLeg />
}`

File: src/SingleBookingPage/bookingItem/AccordionBodyLastLeg.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  leg: AccordionBodyLastLeg_leg,
+|};
+
+const RawAccordionLastLeg = (props: LastLegProps) => {
+  const { leg } = props;
+  const initialDeparture = idx(leg.departure, _ => _.localTime) || '';
+  return (
+    <React.Fragment>
+      <div className="lastLeg">
+        <div className="calendar-icon">
```
Comment (jaroslav-kubicek) [inline src/SingleBookingPage/bookingItem/AccordionBodyLastLeg.js]: please use consistent names for classes `lastLeg` vs `calendar-icon`

File: src/SingleBookingPage/bookingItem/AccordionBodyLeg.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
+// @flow
```
Comment (jaroslav-kubicek) [inline src/SingleBookingPage/bookingItem/AccordionBodyLeg.js]: I would suggest one more thing - let's merge these two components together, `AccordionBodyLastLeg` and `AccordionBodyLastLeg` and pass nextLeg={null} for the last one

btw `Kiwi.com Guarantee` needs to be displayed based on data from GraphQL (new field needs to be added) - so let's just render it everytime now

File: src/SingleBookingPage/bookingItem/AccordionHeader.js (hunk 1 above)
Comment (git-toni) [reply src/SingleBookingPage/bookingItem/AccordionHeader.js]: > then you don't need to pass down toggle callback

Then who is the one toggling the `expanded` state?

File: src/SingleBookingPage/bookingItem/AccordionBodyLeg.js (hunk 4 above)
Comment (git-toni) [reply src/SingleBookingPage/bookingItem/AccordionBodyLeg.js]: > then you don't need to pass down toggle callback

I think the data extracted via their fragments is quite different from one case to the other. If we want to be strict about relay, shouldn't we keep them separate as to not over-request/feed data to the component? (definitely not a critical case, just saying :slightly_smiling_face: )

File: src/helpers/dateUtils.js [hunk 5]
```diff
@@ -35,3 +35,14 @@ export const formatCountDown = (hoursLeft: number) => {
...
   return `${ndays} days`;
 };
+
+export const formatTimeDuration = (mins: number) => {
+  const nhours = Math.floor(mins / 60);
+  const nmins = mins - nhours * 60;
+  return nhours ? `${nhours}h ${nmins}min` : `${nmins}min`;
+};
+
+export const formatHour = (date: string, is12Hour: boolean = true) => {
```
Comment (jaroslav-kubicek) [inline src/helpers/dateUtils.js]: btw looks like `is12Hour` param is not needed and it's true always

File: src/helpers/__tests__/dateUtils.test.js [hunk 6]
```diff
@@ -29,4 +29,18 @@ describe('date utils', () => {
...
+    it('should format duration', () => {
+      expect(dateUtils.formatTimeDuration(90)).toBe('1h 30min');
+      expect(dateUtils.formatTimeDuration(32)).toBe('32min');
+      expect(dateUtils.formatTimeDuration(191)).toBe('3h 11min');
+    });
+  });
+  describe('formatHour', () => {
+    it('should format duration', () => {
+      const myDate = '2018-05-14T13:30:16.905Z';
+      expect(dateUtils.formatHour(myDate)).toBe('3:30 PM');
```
Comment (jaroslav-kubicek) [inline src/helpers/__tests__/dateUtils.test.js]: this isn't correct, it should display `1:30 PM`

CI fails correcly
toLocaleString needs to take `timeZone: 'UTC'` as part of the options

Comment (git-toni) [reply src/helpers/__tests__/dateUtils.test.js]: Yep, you're right. I was making sure all graphql times are UTC-timezoned, and that is the case. Fixing this now, thx.

File: src/SingleBookingPage/bookingItem/AccordionBodyLeg.js (hunk 4 above)
Comment (jaroslav-kubicek) [reply src/SingleBookingPage/bookingItem/AccordionBodyLeg.js]: Now I don't know how it's related to relay 🤔

**Pull request merged**